### PR TITLE
servermaster/election(engine): fix Election goroutine leaks

### LINKED ISSUE #5756: leader context aliveness check goroutine count keeps growing
### What did you do?

Setup a single server master node, observe the goroutine usage via `debug/pprof/goroutine?debug=1` API

### What did you expect to see?

Goroutine count keeps steady.

### What did you see instead?

```go
goroutine profile: total 2118
1602 @ 0x43b516 0x44af92 0x1b77af5 0x46bce1
#	0x1b77af4	github.com/pingcap/tiflow/engine/servermaster/cluster.(*leaderCtx).Done.func1+0x94	github.com/pingcap/tiflow/engine/servermaster/cluster/election.go:170

427 @ 0x43b516 0x44af92 0x19b1958 0x46bce1
#	0x19b1957	github.com/pingcap/tiflow/engine/pkg/errctx.(*errCtx).Done.func1.1+0x77	github.com/pingcap/tiflow/engine/pkg/errctx/ctx.go:41
```

It is not a good choice to create a new goroutine for each call of `leaderCtx.Done()`, and leave the goroutine running.
https://github.com/pingcap/tiflow/blob/84145a17b4ea207125c1b9410458a2b4efd428d5/engine/servermaster/cluster/election.go#L166-L182

### Versions of the cluster

Dataflow Engine version (run `df-master -V` or `df-executor -V`):

master@https://github.com/pingcap/tiflow/commit/84145a17b4ea207125c1b9410458a2b4efd428d5

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?

Issue Number: close #5756 

### What is changed and how it works?
- Create `doneCh` only once using `sync.Once`.
- Removed timeout config for `NewSession` because it is difficult to implement correctly given the current `clientv3` implementation provided by Etcd. The original implementation was not correct.
- Improved unit test coverage.

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No.

##### Do you need to update user documentation, design documentation or monitoring documentation?
No.

### Release note 

```release-note
None.
```

### CODE REVIEW

**Review by amyangfei — APPROVED** [review]

File: engine/servermaster/cluster/election_test.go [hunk 1]
```diff
@@ -111,6 +112,56 @@ func TestEtcdElectionCampaign(t *testing.T) {
...
+
+	select {
+	case <-sessCtx.Done():
+		require.FailNow(t, "sessCtx is not expected to be done by now.")
+	default:
+	}
+
+	require.NoError(t, election.session.Close())
+	<-sessCtx.Done()
+	require.Regexp(t, ".*ErrMasterSessionDone.*", sessCtx.Err())
```
Comment (gozssky) [inline engine/servermaster/cluster/election_test.go:139]:
```suggestion
	require.Regexp(t, "ErrMasterSessionDone", sessCtx.Err())
```

File: engine/servermaster/cluster/election_test.go [hunk 2]
```diff
@@ -111,6 +112,56 @@ func TestEtcdElectionCampaign(t *testing.T) {
...
+func TestCreateSessionFail(t *testing.T) {
+	ctx := context.Background()
+	clientStub := clientv3.NewCtxClient(ctx)
+	clientStub.Lease = &failLease{}
+	_, err := NewEtcdElection(ctx, clientStub, nil, EtcdElectionConfig{
+		TTL:    5,
+		Prefix: "/test-election",
+	})
+	require.Error(t, err)
+	require.Regexp(t, ".*ErrMasterEtcdCreateSessionFail.*", err)
```
Comment (gozssky) [inline engine/servermaster/cluster/election_test.go:162]:
```suggestion
	require.Regexp(t, "ErrMasterEtcdCreateSessionFail", err)
```

**Review by gozssky — APPROVED** [review]
LGTM

There are some tips for `require.Regexp`. See https://github.com/pingcap/tidb/pull/30073#discussion_r755295895.

**Pull request merged**